feat: Add LM Studio Model and Embeddings Component

EDLLT opened the pull request:
Fixes https://github.com/langflow-ai/langflow/issues/3973

ogabrielluiz commented on src/backend/base/langflow/components/models/LMStudioModel.py:
```diff
@@ -0,0 +1,165 @@
+            seed=seed,
+        )
+        if json_mode:
+            if output_schema_dict:
+                output = output.with_structured_output(schema=output_schema_dict, method="json_mode")  # type: ignore
```
I'm using it today (loving it, btw) and it seems it has to be json_schema. We need to update this (and the OpenAI component) to make this work better with BaseModel

EDLLT replied:
> I'm using it today (loving it, btw) and it seems it has to be json_schema. We need to update this (and the OpenAI component) to make this work better with BaseModel

By OpenAI component, you mean ``OpenAIModel.py`` right?

If I understand correctly, all what needs to be done is to swap ``method="json_mode"`` for ``method="json_schema"`` in both ``LMStudioModel.py`` and ``OpenAIModel.py``

ogabrielluiz replied:
Should be

EDLLT replied:
> Should be

I tried that already and it gave me the same error

ogabrielluiz reviewed (commented):
Hey, @EDLLT 

Looking good

EDLLT commented:
> CI error doesn't make sense to me
> 
> ```
> Error: src/backend/base/langflow/components/models/LMStudioModel.py:57:14: RUF005 Consider iterable unpacking instead of concatenation
> Error: Process completed with exit code 1.
> ```
> 
> Line it's complaining about ![image](https://private-user-images.githubusercontent.com/47005435/373697843-fda66f46-bef8-49d8-8051-5126c35ea99d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjgwNTY0MTgsIm5iZiI6MTcyODA1NjExOCwicGF0aCI6Ii80NzAwNTQzNS8zNzM2OTc4NDMtZmRhNjZmNDYtYmVmOC00OWQ4LTgwNTEtNTEyNmMzNWVhOTlkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDA0VDE1MzUxOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVlYTllNTU4ZWViOTdkYmQ5NTIwMjgwNjdhMDQ1Y2FiNTJlNjY5MDllNWQwM2I2OTA5ZDQ3OWYzOTdjMzk5NDgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.razuQqOYC_m1keAnCN0ESMLl8vQ_IRNjvZMkx3QdkN8)
> 
> Every other component I'd seen is structured like that

I guess I'll do what the dosu bot suggested
https://github.com/langflow-ai/langflow/issues/3973#issuecomment-2393976883

ogabrielluiz commented:
Hey @EDLLT 

Something like this: https://github.com/langflow-ai/langflow/blob/54dd81a6812819771029a9a1fa19227fa1fe0ed5/src/backend/base/langflow/components/models/OpenAIModel.py#L30

EDLLT commented:
> Hey @EDLLT 
> 
> Something like this: https://github.com/langflow-ai/langflow/blob/54dd81a6812819771029a9a1fa19227fa1fe0ed5/src/backend/base/langflow/components/models/OpenAIModel.py#L30

Hey, so I've fixed the concatenation issue(dosu bot's pretty useful id say)

All what remains is the json schema comment you had made. 
When turning on the json mode and trying compile the lm studio component, it gives me the same error about json schema. I tried changing json mode and json object to json schema instead but it still gave me the same error.

I'm not completely sure what json mode is supposed to do either so maybe it's not supported by lm studio and should be removed?

EDLLT commented:
@ogabrielluiz Hey. Shall I remove the JSON mode from the lmstudio model component?

EDLLT commented:
Okay. I have removed the JSON mode option completely. 

> @ogabrielluiz Hey. Shall I remove the JSON mode from the lmstudio model component?

Okay, given how it was deprecated anyway, I have removed it.

ogabrielluiz reviewed (approved):
Hey @EDLLT 

Thanks again for this. 

LGTM
